lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1395,6 +1414,45 @@ def cmd_sessions(*args)
           sid = nil
         end
       end
+
+   #launch server that will host meterpreter Web interface
+   # open WebConsole for a perticular sid
+   # start thread for server and open_browser_rtc. msfconsole is already running on the main thread
+   # the webconsole should be visible under sessions -l command in the different section
+    when 'web_ui'
+      print_status("Starting WebConsole the following session(s): #{session_list.join(', ')}")
+      session_list.each do |sess_id|
+        session = framework.sessions.get(sess_id)
+        if session
+          if session.respond_to?(:response_timeout)
+            last_known_timeout = session.response_timeout
+            session.response_timeout = response_timeout
+          end
+          print_status("Starting Web Console for #{sess_id}")
+          begin
+            # need some house keeping, will resolve soon
+            server_bind='127.0.0.1'
+            server_port=3000
+            session_server=Server::Back.new(framework,sid)
+            thr = []
+            thr << framework.threads.spawn("ConsoletoBrowser",true) do
+              Server::WebConsoleServer.run!(:port=>3000)
+            end
+
+            thr << framework.threads.spawn("OpenBrowser",true) do
```

Comment:
Usage of `framework.thread.spawn` is a great way to make sure all threads are tracked.  

Consider how could these be managed to stop/join them when no longer needed?  

Also if made into a `singleton` would it make sense always to start a `Server::WebconsoleServer` and then just pass users on to it when `sessions -w` is called?